### ISSUE TITLE
✨ Installing applications and packages

### ISSUE THREAD

**Issue by vshymanskyy** [vshymanskyy/ViperIDE#42]
Once #41 is ready, it should be rather easy to implement project import:
- [x] Define a simple `package.json` structure. This can be inspired/adapted from mip, node.js, PlatformIO etc.
- [x] Implement the project import mechanism. This is very similar (identical?) to package installation
- [x] ✨ **Project sharing via link** ✨ i.e.:
    https://viper-ide.org/?install=github:emlearn/emlearn-micropython/examples/xor_trees
    - [x] Open in VM by adding `&vm=1`
- [x] Improve UI/UX
  - [x] Display `README.md` from the sample

**Comment by vshymanskyy** [comment]
`package.json` suggestion:

```json
{
  "name": "MySample",
  "version": "0.1.0",
  "description": "Makes everyone happy",
  "author": "Citizen of the Universe",
  "license": "MIT",
  "readme": "https://link.to/README.md",
  "keywords": "json,rest,http,web,esp32",
  "urls": [
    ["fs:main.py", "src/main.py"],
    ["fs:etc/data.dat", "github:org/repo/dir/file.dat"]
  ],
  "deps": [
    ["collections-defaultdict", "latest"],
    "os-path@latest",
    "aiodns"
  ]
}
```

Notes:
- Metadata added. This should be safe - probably it's all skipped by the current infrastructure
- `urls`, `deps` are the same as described in https://docs.micropython.org/en/latest/reference/packages.html
- `urls` changes:
  - Added ability to specify `urls` relative to `package.json`.
    Probably done in https://github.com/micropython/micropython/pull/12477
  - For files starting with `fs:` the target installation path should pwd (`/` or `/flash`) by default
- Added ability to specify deps as single string (TBD, maybe this is unneded)

All of this needs some changes in ViperIDE (and later in `mpremote`)

Comments and ideas are welcome

**Comment by vshymanskyy** [comment]
@RobertBlackhart I think this would be very useful for education scenarios

**Comment by RobertBlackhart** [comment]
I agree. I really like the ability to link using a URL query. I'd be happy to do some work on #41/#42 if you're looking for volunteers, although I can't promise a timeline right now (not that you're asking, just wanted to set expectations).

**Comment by vshymanskyy** [comment]
No, this is completely on me for now. Also, it's in discussion/definition phase (also setting expectations here :smile:  )

**Comment by vshymanskyy** [comment]
There's now an experimental support of installing applications.
For example you can install [`aiodns` example](https://github.com/vshymanskyy/aiodns/tree/main/examples/simple) using this button:

[<img src="https://raw.githubusercontent.com/vshymanskyy/ViperIDE/refs/heads/main/assets/btn_run.png" alt="Run using ViperIDE" height="42"/>](https://viper-ide.org/?install=github:vshymanskyy/aiodns/examples/simple)

**Comment by jonnor** [comment]
I am adding Run in ViperIDE to another example in emlearn-micropython. Ref this package.json, https://github.com/emlearn/emlearn-micropython/blob/master/examples/har_trees/package.json
However, this example one depends on a couple of libraries which are not in micropython-lib, and fails to install because the dependencies are not known.
The libraries in question are:
- https://github.com/jonnor/micropython-zipfile
- https://github.com/jonnor/micropython-npyfile
And I also plan to utilize these also soon:
- https://github.com/jonnor/micropython-mpu6886
- https://github.com/jonnor/bma423-pure-mp

How to add these?

**Comment by jonnor** [comment]
> How to add these?

Using the URL works! For example `"deps": [ [ "https://github.com/jonnor/micropython-mpu6886", "master" ] ]`. Don't know why I did not try it before...
